src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -85,7 +89,11 @@ def handleHover (p : HoverParams)
       let stxDoc? ← match stack? with
         | some stack => stack.findSomeM? fun (stx, _) => do
           let .node _ kind _ := stx | pure none
-          return (← findDocString? snap.env kind).map (·, stx.getRange?.get!)
+          -- If the tactic is an alternative form, get the docs for the canonical version
+          let kind := alternativeOfTactic snap.env kind |>.getD kind
+          let exts := getTacticExtensionString snap.env kind
+          let docStr := (← findDocString? snap.env kind).map (· ++ exts)
+          return docStr.map (·, stx.getRange?.get!)
```

Comment:
Wouldn't it be easier to copy the docstring to the alternative, like `inherit_doc` does?

Reply:
Because the set of extensions is open-ended, we'd need to also either do the same work of resolving alternatives to get them, or we'd need to copy them over both when extensions are added and when alternatives are added, which strikes me as a trickier invariant to maintain.

Is the concern here that the table lookup will cause latency issues?

Reply:
Ah, it makes sense for extensions of course. I was mostly just worried about adding such specific code to two places in the server. Can we factor it out, perhaps even put it directly below `findDocString?`?

Reply:
That in particular leads to circular module imports, unfortunately, but factoring it out is a good idea.

I'll go make a module that can be downstream of things that are themselves downstream of the docstring environment extension, and upstream of the server, to put this into.